### DISCUSSION TITLE
BUG: array fields are not correctly described when you use the "Definition" option to display a table in SQL

### DISCUSSION THREAD

**Discussion by peppermintbunny** [supabase/supabase#13924] [General]
Bug found in the new "Definition" option which display's a given table's structure in SQL

If one uses the new "Definition" option to display a table's structure in SQL there is an error in the SQL. Array datatypes are not converted correctly. Their datatype is omitted. So, for example a table field named PURCHASES which is defined in the Table Editor as being of TEXT ARRAY is converted to SQL merely as " PURCHASES ARRAY...." The datatype is not reported.

**Comment by Franco98404** [comment]
I have the same problem

**Comment by Franco98404** [comment]
Related to this I enabled the citext extension and go to the Supabase definition and appears this

```plpgsql
create table
  public.schools (
    school_id uuid not null default uuid_generate_v4 (),
    school_name public.citext not null,
    constraint schools_pk primary key (school_id),
    constraint schools_uq unique (school_name)
  ) tablespace pg_default;
``` 

But when I execute the definition I have this error

```
Failed to run sql query: type "public.citext" does not exist
``` 

So the definition should be `extensions.citext` instead `public.citext`

**Reply by peppermintbunny** [reply]
Interesting.  Hope supabase fix this soon.

**Comment by Gramaz** [comment]
Might this explain why I keep getting a “Value is an invalid array” error on my UUID array when inputting info?

**Comment by alaister** [comment]
Hi @peppermintbunny,

We just shipped a fix for this, so please give it another try!

Thanks for reporting 🙏🏻

**Comment by peppermintbunny** [comment]
Thank you very much for listening!

Charlotte Gamsu
